src/ToolBox/SOS/Strike/strike.cpp

```diff
@@ -12112,6 +12113,71 @@ class ClrStackImpl
         }
 #endif // _TARGET_WIN64_
     }
+    
+    static HRESULT PrintManagedFrameContext(IXCLRDataStackWalk *pStackWalk)
+    {
+        CROSS_PLATFORM_CONTEXT context;
+        HRESULT hr = pStackWalk->GetContext(DT_CONTEXT_FULL, g_targetMachine->GetContextSize(), NULL, (BYTE *)&context);
+        if (FAILED(hr) || hr == S_FALSE)
+        {
+            // GetFrameContext returns S_FALSE if the frame iterator is invalid.  That's basically an error for us.
+            ExtOut("GetFrameContext failed: %lx\n", hr);
+            return E_FAIL;
+        }
+                     
+#if defined(SOS_TARGET_AMD64)
+        String outputFormat3 = "    %-3s=%016x %-3s=%016x %-3s=%016x\n";
+        String outputFormat2 = "    %-3s=%016x %-3s=%016x\n";
+        String outputFormat1 = "    %-3s=%016x\n";
+        ExtOut(outputFormat3, "rax", context.Amd64Context.Rax, "rbx", context.Amd64Context.Rbx, "rcx", context.Amd64Context.Rcx);
+        ExtOut(outputFormat3, "rdx", context.Amd64Context.Rdx, "rsp", context.Amd64Context.Rsp, "rbp", context.Amd64Context.Rbp);
+        ExtOut(outputFormat3, "rip", context.Amd64Context.Rip, "rsi", context.Amd64Context.Rsi, "rdi", context.Amd64Context.Rdi);
+        ExtOut(outputFormat3, "r08", context.Amd64Context.R8, "r09", context.Amd64Context.R9, "r10", context.Amd64Context.R10);
```

Comment:
I also wonder about the register ordering. It seems it might make sense to put rsp, rbp and rip together at the beginning instead of sticking them in the middle.